File: palindromeAnagram.py
'''

We have given a anagram string and we have to check whether it can be made palindrome o not.

Examples:

Input : abcd 
Output : False

Explaination: 
There is no palindrome anagram of 
given string

Input  : geeksgeeks
Output : Yes

Explaination:
There are palindrome anagrams of
given string. For example kgeeseegk
'''

#AUTHOR: SANTOSH D KOLUR




def anagram(n):
    d={c:0 for c in set(list(n))}

    for ele in list(n):
        d[ele]+=1
    occr=list(d.values())
    even_occr=0
    for ele in occr:
        if ele%2==0:
            even_occr+=1
    if(even_occr>=len(occr)-1 and occr.count(1)<=1):
        print("True")
    else:
        print("False")

if __name__== '__main__':
    n=input()
    anagram(n)

File: palindromeAnagram.java

/*
 * We have given a anagram string and we have to check whether it can be made palindrome o not.
 * Examples:

Input : geeksforgeeks 
Output : No
There is no palindrome anagram of 
given string

Input  : geeksgeeks
Output : Yes
There are palindrome anagrams of
given string. For example kgeeseegk
 */

// Author: Nisarg Devdhar
import java.util.Collection;
import java.util.HashMap;
import java.util.Scanner;

public class palindromeAnagram {
	static HashMap<Character,Integer> map=new HashMap<>();
	public static void main(String[] args) {
		System.out.print("String=");
		Scanner sc=new Scanner(System.in);
		String s=sc.next();
		
		//char[] ch=s.toCharArray();
		
		for(int i=0;i<s.length();i++) {
			char ch=s.charAt(i);
			map.put(ch,map.getOrDefault(ch,0)+1);
		}
		Collection<Integer> n=map.values();
		
		int odd=0;
		
		for(Integer i: n) {
			if(i%2!=0)
				odd++;
		}
		
		if(odd>1) {
			System.out.println("false");
			return;
		}
		
		System.out.println("True");
		
	}
}
